app/MyApp/app/views/takeTestView.js

```diff
@@ -164,20 +164,43 @@ $(function() {
                 this.$el.append('<div class="quizActions" ><div class="btn btn-info" id="finishPressed">'+App.languageDict.attributes.Finish+'</div></div>')
                 var sstatus = this.myModel.get('stepsStatus')
                 var sp = this.myModel.get('stepsResult')
+                var stepid = this.myModel.get('stepsIds')
                 var pqattemptss = this.myModel.get('pqAttempts')
                 var flagAttempts = false;
                 flagAttempts = true;
 
                 if(flagAttempts && this.myModel.get('pqAttempts')) {
                     var pqattempts = this.myModel.get('pqAttempts')
                 }
-                if (pqattempts != undefined) {
-                   if( pqattempts[this.stepindex].length > 1) {
-                     pqattempts[this.stepindex][1]++;
-                   } else {
-                        pqattempts[this.stepindex]++;
-                   }
-                   this.myModel.set('pqAttempts', pqattempts)
+                if( sstatus[this.stepindex][pqattemptss[this.stepindex]] == null){
+                    var courseAnswer = new App.Collections.CourseAnswer()
+                    courseAnswer.MemberID = $.cookie('Member._id')
+                    courseAnswer.StepID = stepid[this.stepindex]
+                    courseAnswer.pqattempts = pqattempts[this.stepindex]
+                    courseAnswer.fetch({
+                        async: false
+                    })
+                    var answerLength = courseAnswer.models.length-1;
+                    for (var j = answerLength; j >= 0; j--) {
+                        courseAnswer.models[j].destroy();
+                    }
+                    if (pqattempts != undefined) {
+                       if( pqattempts[this.stepindex].length > 1) {
+                         pqattempts[this.stepindex][1];
+                       } else {
+                            pqattempts[this.stepindex];
+                       }
+                       this.myModel.set('pqAttempts', pqattempts)
+                    }
+                } else {
+                    if (pqattempts != undefined) {
+                        if( pqattempts[this.stepindex].length > 1) {
+                            pqattempts[this.stepindex][1]++;
+                        } else {
```

Comment:
these were code from old course structure with paper and test so this condition is not required in new structure